From b8eb9b993185ae9598cfd6d075fc42bbd1500dce Mon Sep 17 00:00:00 2001
From: Tony Cheng <tony.cheng@amd.com>
Date: Sat, 17 Sep 2016 15:33:28 -0400
Subject: [PATCH 1103/1722] drm/amd/dal: refactor aux 4

- remove dce_11_0_d.h dependancy, as offsets are passed in from upper layer.
- remove register offset related code

Change-Id: Ice01329def98c7d37757c63ffcd097ead2aa8f7a
Signed-off-by: Tony Cheng <tony.cheng@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 .../amd/dal/dc/i2caux/dce110/aux_engine_dce110.c   | 26 ----------------------
 1 file changed, 26 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
index e280ed2..88db273 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
@@ -41,7 +41,6 @@
 /*
  * Post-requisites: headers required by this unit
  */
-#include "dce/dce_11_0_d.h"
 #include "dce/dce_11_0_sh_mask.h"
 
 #define CTX \
@@ -395,15 +394,6 @@ static enum aux_channel_operation_result get_channel_status(
 	}
 }
 
-static const int32_t aux_channel_offset[] = {
-	mmDP_AUX0_AUX_CONTROL - mmDP_AUX0_AUX_CONTROL,
-	mmDP_AUX1_AUX_CONTROL - mmDP_AUX0_AUX_CONTROL,
-	mmDP_AUX2_AUX_CONTROL - mmDP_AUX0_AUX_CONTROL,
-	mmDP_AUX3_AUX_CONTROL - mmDP_AUX0_AUX_CONTROL,
-	mmDP_AUX4_AUX_CONTROL - mmDP_AUX0_AUX_CONTROL,
-	mmDP_AUX5_AUX_CONTROL - mmDP_AUX0_AUX_CONTROL
-};
-
 static const struct aux_engine_funcs aux_engine_funcs = {
 	.destroy = destroy,
 	.acquire_engine = acquire_engine,
@@ -423,14 +413,6 @@ static bool construct(
 	struct aux_engine_dce110 *engine,
 	const struct aux_engine_dce110_init_data *aux_init_data)
 {
-	int32_t offset;
-
-	if (aux_init_data->engine_id >=
-		sizeof(aux_channel_offset) / sizeof(int32_t)) {
-		ASSERT_CRITICAL(false);
-		return false;
-	}
-
 	if (!dal_aux_engine_construct(
 		&engine->base, aux_init_data->ctx)) {
 		ASSERT_CRITICAL(false);
@@ -438,16 +420,8 @@ static bool construct(
 	}
 	engine->base.base.funcs = &engine_funcs;
 	engine->base.funcs = &aux_engine_funcs;
-	offset = aux_channel_offset[aux_init_data->engine_id];
-	engine->addr.aux_control = mmAUX_CONTROL + offset;
-	engine->addr.aux_arb_control = mmAUX_ARB_CONTROL + offset;
-	engine->addr.aux_sw_data = mmAUX_SW_DATA + offset;
-	engine->addr.aux_sw_control = mmAUX_SW_CONTROL + offset;
-	engine->addr.aux_interrupt_control = mmAUX_INTERRUPT_CONTROL + offset;
-	engine->addr.aux_sw_status = mmAUX_SW_STATUS + offset;
 
 	engine->timeout_period = aux_init_data->timeout_period;
-
 	engine->regs = aux_init_data->regs;
 
 	return true;
-- 
2.7.4

